src/simulator.zig

```diff
@@ -451,13 +451,14 @@ pub const Simulator = struct {
         }
 
         // Even though there are no client requests in progress, the cluster may be upgrading.
-        const release_max = simulator.core_release_max();
```

Comment:
Will this work? Given that core replicas don't restart normally, I think they aren't guaranteed to progress to `replica_releases_limit - 1`?

Reply:
Ahh you're right, I forgot that `transition_to_liveness_mode` zeroes the `replica_release_advance_probability`!
So it is possible that 1/6 replicas (outside the core) has version 3, and all other replicas have version 2. That would get stuck in this case.